### DISCUSSION TITLE
config `delegateEvents` all in runtime

### DISCUSSION THREAD

**Discussion by zhaoyao91** [solidjs/solid#464] [Ideas]
from this [issue](https://github.com/ryansolid/dom-expressions/issues/48) and this [issue](https://github.com/solidjs/solid/issues/334)

The separated config for delegateEvents in both compile and runtime is not good. We make components library using JSX, but we have to bind half of delegateEvents in its babel config, with the other half of delegateEvents leaving to the app developer to config it at runtime.

I thought for tha babel plugin, it may designate the task of **decide how to bind events** to the solid library, and let the library decide how to bind events by runtime config.

**Comment by ryansolid** [comment]
It's not that simple because the babel plugin happens at compile-time, the compiled output differs depending on it. There would need to be some sort of shared config that the runtime tapped into that was prepared at compile time. But the runtime can't drive this without de-opting everything.

**Reply by zhaoyao91** [reply]
Yeah, babel at complie-time. I mean, in compile-time, we do not write code adding listener directly, instead, we call some function from solid-js, say, `registerListener`, so, how to actually add lisenter becomes a decision of solid-js at runtime.

This is just my rough thought. I'll try to look at the source code for details later.

**Reply by zhaoyao91** [reply]
Wow, after searching the codebase, I found it's already the case I wish. Whatever `delegateEvents` is at the compile-time, if we `DelegatedEvents.clear()` them on the runtime, all events won't be delegated. Good.
